### PR TITLE
Fix test retry

### PR DESCRIPTION
**Pull request by drwill-ms**
Bad logic causing retry not to happen when device not found due to slow device registry.

### CODE REVIEW

File: e2e/test/helpers/HubServiceTestRetryPolicy.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public bool ShouldRetry(uint currentRetryCount, Exception lastException, out Tim
         {
             retryInterval = TimeSpan.Zero;
 
-            if (currentRetryCount < MaxRetries)
+            if (currentRetryCount > MaxRetries)
```
Comment (drwill-ms) [inline e2e/test/helpers/HubServiceTestRetryPolicy.cs:25]: This was the oopsie.

File: e2e/test/helpers/TestDevice.cs [hunk 2]
```diff
@@ -52,25 +48,14 @@ private TestDevice(Device device, IAuthenticationMethod authenticationMethod)
         /// <param name="type">The way the device will authenticate</param>
         public static async Task<TestDevice> GetTestDeviceAsync(string namePrefix, TestDeviceType type = TestDeviceType.Sasl)
         {
-            string prefix = namePrefix + type + "_";
-
-            try
-            {
-                await s_semaphore.WaitAsync().ConfigureAwait(false);
```
Comment (drwill-ms) [inline e2e/test/helpers/TestDevice.cs:59]: Why do we have a semaphore here??

Comment (timtay-microsoft) [reply e2e/test/helpers/TestDevice.cs:59]: The service client operations needed to create a new test device should already be thread safe, so I agree with removing this semaphore

**Review by timtay-microsoft — APPROVED** [review]

File: e2e/test/helpers/TestDevice.cs (hunk 2 above)
Comment (timtay-microsoft) [reply e2e/test/helpers/TestDevice.cs:59]: It's just an HTTP request, after all

**Review by brycewang-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

**Review by andyk-ms — APPROVED** [review]

**Pull request merged**